src/main/java/de/tum/in/www1/artemis/service/ExerciseService.java

```diff
@@ -722,4 +722,13 @@ public List<Feedback> getFeedbackToBeDeletedAfterGradingInstructionUpdate(boolea
         }
         return feedbackToBeDeleted;
     }
+
+    /**
+     * To avoid direct access to the {@link ExerciseRepository}, we use delegation to find the corresponding {@link Exercise} based on its id.
+     * @param exerciseId id of the {@link Exercise}
+     * @return Optional<Exercise>
+     */
+    public Optional<Exercise> findById(Long exerciseId) {
```

Comment:
Could you maybe explain the rationale behind this a bit? Elsewhere in Artemis we simply directly access the ExerciseRepository when we need to load an exercise by its id.
Unless there is a good reason for this proxy method, I don't think you should deviate from this, as this would only lead to unnecessary methods in services.

Reply:
Sure. Normally you should avoid working directly on the repositories as this is not really a good practice. I think we tried to do this in Artemis as well, but some reviewers were not so strict to enforce this behavior :thinking: (and if we try to enforce this behavior now, we get some problems with circular dependencies).

Reply:
Solved via PM :+1: